hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/JobMetrics_StreamTest.java

```diff
@@ -0,0 +1,287 @@
+import com.hazelcast.jet.config.ProcessingGuarantee;
+import com.hazelcast.jet.core.JobMetrics;
+import com.hazelcast.jet.pipeline.JournalInitialPosition;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.jet.pipeline.Sources;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.hazelcast.jet.core.JobStatus.FAILED;
+import static com.hazelcast.jet.core.JobStatus.RUNNING;
+import static com.hazelcast.jet.core.JobStatus.SUSPENDED;
+import static org.junit.Assert.assertEquals;
+
+
+public class JobMetrics_StreamTest extends TestInClusterSupport {
+
+    private static final String NOT_FILTER_OUT_PREFIX = "ok";
+    private static final String FILTER_OUT_PREFIX = "nok";
+
+    private static final String FLAT_MAP_AND_FILTER_VERTEX = "fused(map, filter)";
+    private static final String RECEIVE_COUNT_METRIC = "receivedCount";
+    private static final String EMITTED_COUNT_METRIC = "emittedCount";
+
+    private static String journalMapName;
+    private static String sinkListName;
+
+    @Before
+    public void before() {
+        journalMapName = JOURNALED_MAP_PREFIX + randomString();
+        sinkListName = "sinkList" + randomString();
+    }
+
+    @Test
+    public void when_jobRunning_then_metricsEventuallyExist() {
+        Map<String, String> map = testMode.getJet().getMap(journalMapName);
+        putIntoMap(map, 2, 1);
+        List<String> sink = testMode.getJet().getList(sinkListName);
+
+        Pipeline p = createPipeline();
+        // When
+        Job job = testMode.getJet().newJob(p);
+
+        assertTrueEventually(() -> assertEquals(2, sink.size()));
+        // Then
+        assertTrueEventually(() -> assertMetrics(job.getMetrics(), 3, 1));
+
+        putIntoMap(map, 1, 1);
+        assertTrueEventually(() -> assertEquals(3, sink.size()));
+        // Then
+        assertTrueEventually(() -> assertMetrics(job.getMetrics(), 5, 2));
+    }
+
+    @Test
+    public void when_jobCancelled_then_terminalMetricsExist() {
+        Map<String, String> map = testMode.getJet().getMap(journalMapName);
+        putIntoMap(map, 2, 1);
+        List<String> sink = testMode.getJet().getList(sinkListName);
+
+        Pipeline p = createPipeline();
+        Job job = testMode.getJet().newJob(p);
+
+        putIntoMap(map, 1, 1);
+
+        assertTrueEventually(() -> assertEquals(3, sink.size()));
+
+        // When
+        job.cancel();
+        assertJobStatusEventually(job, FAILED);
+        // Then
+        assertTrueEventually(() -> assertMetrics(job.getMetrics(), 5, 2));
+    }
+
+    @Test
+    @Ignore("https://hazelcast.atlassian.net/wiki/spaces/JET/pages/1758560997/"
+            + "Metrics+for+Completed+Jobs+Design?focusedCommentId=1767506913#comment-1767506913")
+    public void metricsExistWhenJobSuspendedAndResumed() {
```

Comment:
I'm for removing this test. We should not try to preserve metrics from previous execution. Metrics are local for a member and members can come and go, how would we repartition the metrics?

Reply:
I've update this test to be aligned with current behaviour.